streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImplJoin.java

```diff
@@ -173,7 +174,7 @@ public <K, V1, V2, VOut> KStream<K, VOut> join(final KStream<K, V1> lhs,
 
         final JoinWindowsInternal internalWindows = new JoinWindowsInternal(windows);
         final KStreamKStreamJoin<K, V1, V2, VOut> joinThis = new KStreamKStreamJoin<>(
-            true,
+            JoinSide.LEFT,
```

Comment:
I think these call-sites are much more clear with the explicit "left" and "right". It's interesting that the variables here are joinThis and joinOther, instead of joinLeft and joinRight.